elide-async/src/main/java/com/yahoo/elide/async/service/AsyncQueryThread.java

```diff
@@ -46,63 +54,195 @@
+     */
+    protected Integer calculateRecordsJSON(String jsonStr) {
+        Integer rec;
+        try {
+            JSONObject j = new JSONObject(jsonStr);
+            rec = j.getJSONArray("data").length();
+
+        } catch (JSONException e) {
+            rec = null;
+        }
+        return rec;
+    }
+
+    /**
+     * This method calculates the number of records from the response with GRAPHQL API.
+     * @param response is the response.getBody() we get from the response
+     * @param table_name is the table from which we extract the data
+     * @return rec is the recordCount
+     */
+    protected Integer calculateRecordsGraphQL(String response, String table_name) {
+        Integer rec;
+        try {
+            JSONObject j = new JSONObject(response);
+            JSONObject j2 = j.getJSONObject("data");
+            JSONObject j3 = j2.getJSONObject(table_name);
+            rec = j3.getJSONArray("edges").length();
+
+        } catch (JSONException e) {
+            rec = null;
+        }
+        return rec;
+    }
+
+    /**
+     * This method helps to extract the table name from the query.
+     * @param jsonStr is the query with which we are extracting the data
+     * @return the table name from the above query
+     */
+    protected String getTableNameFromQuery(String jsonStr) {
+        StringBuilder str = new StringBuilder();
+        try {
+            JSONObject j = new JSONObject(jsonStr);
+            String s = (String) j.get("query");
+
+            Integer countBrackets = 0;
+            for (int i = 0; i < s.length(); i++) {
+                if (s.charAt(i) == '{' || s.charAt(i) == '}') {
+                    countBrackets++;
+                    if (countBrackets == 2) {
+                        break;
+                    }
+                }
+                else {
+                    str.append(s.charAt(i));
+                }
+            }
+        } catch (JSONException e) {
+            log.error("Exception: {}", e);
+
+        }
+        return str.toString().trim().split("\\s+")[0];
+    }
+
+    /**
+     * This method converts the JSON response to a CSV response type.
+     * @param jsonStr is the response.getBody() of the query
+     * @return retuns a string which nis in CSV format
+     */
+    protected String convertJsonToCSV(String jsonStr) {
+        if (jsonStr == null) {
+            return null;
+        }
+        StringBuilder str = new StringBuilder();
+
+        try {
+            new JSONObject(jsonStr);
+            JFlat flatMe = new JFlat(jsonStr);
+            List<Object[]> json2csv = flatMe.json2Sheet().headerSeparator("_").getJsonAsSheet();
+
+            for (Object[] obj : json2csv) {
+                str.append(Arrays.toString(obj));
+            }
+
+        } catch (JSONException e) {
+            log.error("Exception: {}", e);
+        } catch (PathNotFoundException e) {
+            log.error("Exception: {}", e);
+        } catch (Exception e) {
```

Comment:
What exception are we trying to capture here?  I'm worried about swallowing NPE and other fatal errors we may want to let propagate.